### PR TITLE
Create a QA Release workflow for Flex plugins

### PR DESCRIPTION
**Pull request by GPaoloni**
Primary reviewer: @stephenhand might be the one with more context here, but feel free to defer (please let me know if I should tag someone else).

## Description
This PR adds a new workflow, "Pre-release QA", that asks for a branch, a prefix and a description and will:
- Deploy the branch to e2e Flex account.
- Run e2e tests on it.
- Creates the "next iteration" pre-release for the given prefix. If user provides `v1.2.3` as prefix, then this workflow will:
  - Create `v1.2.3-qa.1` if this is the first time the worklfow is called with this prefix or,
  - Create `v1.2.3-qa.<x + 1>` if `v1.2.3-qa.<x>` already exists.
- Sends a Slack message if everything succeeds :rocket:.

This is the result of running twice the workflow with the prefix `v0.0.2`:
- First run created https://github.com/techmatters/flex-plugins/releases/tag/v0.0.2-qa.1.
- Second run created https://github.com/techmatters/flex-plugins/releases/tag/v0.0.2-qa.2.

Screenshot from 2022-08-29 18-24-52

~Note: There is a lot of room to improve the changelog, I'm still not sure how that works, but _I believe_ it matches all the commits since the last tag on the branch?~

### Checklist
- [x] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-1262)
- [n/a] New tests added
- [n/a] Strings are localized
- [n/a] Tested for chat contacts
- [n/a] Tested for call contacts

### Verification steps
Follow [this example](https://stackoverflow.com/questions/63362126/github-actions-how-to-run-a-workflow-created-on-a-non-master-branch-from-the-wo/67840292#67840292) to trigger at least one time using `v0.0.2` prefix, or twice with a different one.

### CODE REVIEW

File: .github/workflows/plugin-hrm-form-pre-release-qa.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      - uses: "marvinpinto/action-automatic-releases@v1.2.1"
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          automatic_release_tag: ${{ steps.generate_next_tag.outputs.generated-tag }}
+          prerelease: true
+          title: ${{ inputs.title }}
+          # files: |
+          #   LICENSE.txt
+          #   *.jar
+        id: create_pre_release
```
Comment (GPaoloni) [inline .github/workflows/plugin-hrm-form-pre-release-qa.yml:96]: We could specify more specific files, rn it's bundling the whole repo.

**Comment by GPaoloni** [comment]
The last attempts running this failed the E2E part, I would love to merge against master and then re-iterate, since I'm not sure if the issue here is with the tests being run from the branch. I have merged master against it but still causing the same problems..

**Comment by GPaoloni** [comment]
(For future me)

After merging this PR I want to:
- Check if the E2E tests are failing because of the branch.
- Use the existing deploy workflow for E2E to deploy which will reduce some duplication here (like in https://github.com/techmatters/serverless/pull/302).
- Factor out the entire "generate new tag" action so we can DRY that and use here and in https://github.com/techmatters/serverless/pull/302.

File: .github/workflows/plugin-hrm-form-pre-release-qa.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        id: latest_matching_tag
+
+      # Upload artifacts
+      - uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: test-artifacts
+          path: e2e-tests/test-results
+
+      ## TODO: test if this is aborted if E2E tests fails
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply .github/workflows/plugin-hrm-form-pre-release-qa.yml]: Yes! Removing old comment :P

File: .github/actions/generate-next-tag/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/**
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply .github/actions/generate-next-tag/index.js]: Changed to be an npm script, thanks for the suggestion!

File: .github/workflows/plugin-hrm-form-pre-release-qa.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        uses: ./.github/actions/generate-next-tag
+        with:
+          prefix: "${{ inputs.tag-prefix }}-qa"
+          latest-matching-tag: ${{ steps.latest_matching_tag.outputs.tag }}
+        id: generate_next_tag
+
+      - run: echo ${{ steps.generate_next_tag.outputs.generated-tag }}
+
+      # Create a pre-release
+      - uses: "marvinpinto/action-automatic-releases@v1.2.1"
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply .github/workflows/plugin-hrm-form-pre-release-qa.yml]: Yes you are right. I believe we'll want to create our own for this. Are we fine deferring that for a next ticket? It's probably the biggest action in this flow.

File: .github/workflows/plugin-hrm-form-pre-release-qa.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        required: true
+      title:
+        description: Release title - The title that will be given to this pre-release.
+        required: true
+
+jobs:
+
+  build-and-deploy:
+    # TODO: remove branch tag
+    uses: techmatters/flex-plugins/.github/workflows/end_to_end_development.yml@gian_CHI-1262-p2
```
Comment (stephenhand) [inline .github/workflows/plugin-hrm-form-pre-release-qa.yml:20]: I assume the branch needs to be updated in a separate PR?

**Review by stephenhand — APPROVED** [review]
LGTM, just 1 q

File: .github/workflows/plugin-hrm-form-pre-release-qa.yml (hunk 5 above)
Comment (GPaoloni) [reply .github/workflows/plugin-hrm-form-pre-release-qa.yml:20]: I will do this in 1 more commit, before merging this. Thanks for the reminder I almost go n merge when saw the approved hehe.
